### PR TITLE
core: Add aformat_messages to FewShotChatMessagePromptTemplate and ChatPromptTemplate

### PR DESCRIPTION
**Pull request by cbornet**
Needed since the example selector may use a vector store.

### CODE REVIEW

File: libs/core/langchain_core/prompts/chat.py [hunk 1]
```diff
@@ -901,19 +929,31 @@ def from_messages(
             partial_variables=partial_vars,
         )
 
-    def format(self, **kwargs: Any) -> str:
```
Comment (cbornet) [inline libs/core/langchain_core/prompts/chat.py:904]: Removed `format` from `ChatPromptTemplate` as it is identical to the one in the parent class `BaseChatPromptTemplate `

File: libs/core/langchain_core/prompts/chat.py [hunk 2]
```diff
@@ -66,6 +67,17 @@ def format_messages(self, **kwargs: Any) -> List[BaseMessage]:
...
+    async def aformat_messages(self, **kwargs: Any) -> List[BaseMessage]:
+        """Format messages from kwargs. Should return a list of BaseMessages.
+
+        Args:
+            **kwargs: Keyword arguments to use for formatting.
+
+        Returns:
+            List of BaseMessages.
+        """
+        return await run_in_executor(None, self.format_messages, **kwargs)
```
Comment (eyurtsev) [inline libs/core/langchain_core/prompts/chat.py:79]: This is another place where I feel like we need to delegate to the sync implementation.

I'm a bit worried about having this dichotomy / how to make it obvious that this is is what's going on

Comment (eyurtsev) [reply libs/core/langchain_core/prompts/chat.py:79]: cc @baskaryan  / @nfcampos 

Do we keep run_in_executor for consistency or use a blocking call to the sync code?

Comment (eyurtsev) [reply libs/core/langchain_core/prompts/chat.py:79]: Another PR where this question comes up: https://github.com/langchain-ai/langchain/pull/19652/files

Comment (eyurtsev) [reply libs/core/langchain_core/prompts/chat.py:79]: This is the part that makes it all weird:

https://github.com/langchain-ai/langchain/blob/2e0ddd6fb805af53d9bfb0edc463543a0312530f/libs/core/langchain_core/prompts/chat.py#L934-L934

Chat prompt templates are constructed of multiple messages each of which can be formatted independently -- this 
would mean that interpolating a few variables into essentially a list of dicts we're launching several threads

Comment (cbornet) [reply libs/core/langchain_core/prompts/chat.py:79]: Another possibility if we don't want to choose now is to raise NotImplementedError in the default interface.

Comment (cbornet) [reply libs/core/langchain_core/prompts/chat.py:79]: I intend to implement all subclasses in the end so the default interface won't be used anyway.

Comment (cbornet) [reply libs/core/langchain_core/prompts/chat.py:79]: > a list of dicts we're launching several threads

Actually, it's not launching several threads, it's submitting tasks to the default thread-pool (which has 1 thread per core).

Comment (cbornet) [reply libs/core/langchain_core/prompts/chat.py]: Anyway, I updated the PR to call the sync method directly. PTAL

**Review by eyurtsev — APPROVED** [review]

**Pull request merged**